.github/workflows/release.yml

```diff
@@ -0,0 +1,57 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  verify-branch:
+    name: Verify that runs on the main branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Fail if branch is not main
+        if: github.ref != 'refs/heads/main'
+        run: |
+          echo "The release workflow should only be triggered on the main branch"
+          exit 1
+
+  get-version:
+    name: Get version from Cargo.toml
+    needs: verify-branch
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get version from Cargo.toml
+        id: lookupVersion
+        uses: mikefarah/yq@f15500b20a1c991c8729870ba60a4dc3524b6a94
+        with:
+          cmd: yq -oy '"v" + .workspace.package.version' 'Cargo.toml'
+
+      - name: Print version
```

Comment:
should we verify the version does not yet exist as well?

Reply:
This is really good idea. Added `verify-tag-existence` job